stdlib/src/utils/string_slice.mojo

```diff
@@ -312,3 +313,45 @@ struct StringSlice[
         without the string getting deallocated early.
         """
         pass
+
+    fn isspace(self) -> Bool:
+        """Determines whether the given StringSlice is a python
+        whitespace String. This corresponds to Python's
+        [universal separators](
+            https://docs.python.org/3/library/stdtypes.html#str.splitlines)
+        `" \\t\\n\\r\\f\\v\\x1c\\x1d\\x1e\\x85\\u2028\\u2029"`.
+
+        Returns:
+            True if the String is one of the whitespace characters
+                listed above, otherwise False.
+        """
+        # TODO add line and paragraph separator as stringliteral
+        # once unicode escape secuences are accepted
+        var next_line = List[UInt8](0xC2, 0x85)
+        """TODO: \\x85"""
+        var unicode_line_sep = List[UInt8](0xE2, 0x80, 0xA8)
+        """TODO: \\u2028"""
+        var unicode_paragraph_sep = List[UInt8](0xE2, 0x80, 0xA9)
+        """TODO: \\u2029"""
+
+        @always_inline
+        fn _compare(
+            item1: UnsafePointer[UInt8], item2: UnsafePointer[UInt8], amnt: Int
+        ) -> Bool:
+            var ptr1 = DTypePointer(item1)
+            var ptr2 = DTypePointer(item2)
+            return memcmp(ptr1, ptr2, amnt) == 0
+
+        var no_null_len = len(self)
+        var ptr = self.unsafe_ptr()
+        if no_null_len == 1 and _isspace(ptr[0]):
+            return True
+        elif no_null_len == 2 and _compare(ptr, next_line.unsafe_ptr(), 2):
+            return True
+        elif no_null_len == 3 and (
+            _compare(ptr, unicode_line_sep.unsafe_ptr(), 3)
+            or _compare(ptr, unicode_paragraph_sep.unsafe_ptr(), 3)
+        ):
+            return True
+        _ = next_line, unicode_line_sep, unicode_paragraph_sep
+        return False
```

Comment:
This implementation only works for a single unicode character, right? It would be better to not have that restriction, given that String doesn't have it either.

Reply:
oh so you mean moving the whole implementation into `StringSlice.isspace()`? That would mean having to iterate over the `StringSlice` checking for utf8 byte sequence type etc. , so basically moving/copying the `_StringIter` implementation over to `_StringSliceIter` and using that ? Some small overhead might be incurred when someone first calls String.__iter__ and then does .isspace() on the StringSlice though. Should I do that anyway?

Reply:
It would be better to do that right away. It's a pattern we're trying to move to: implement for `StringSlice` and delegate from `String`. It will also reduce the amount of churn we have to do internally to review and land PRs. Thanks!